lib/css/CssModulesPlugin.js

```diff
@@ -125,53 +126,51 @@ class CssModulesPlugin {
 					StaticExportsDependency,
 					new StaticExportsDependency.Template()
 				);
-				normalModuleFactory.hooks.createParser
-					.for("css")
-					.tap(plugin, parserOptions => {
-						validateParserOptions(parserOptions);
-						return new CssParser();
-					});
-				normalModuleFactory.hooks.createParser
-					.for("css/global")
-					.tap(plugin, parserOptions => {
-						validateParserOptions(parserOptions);
-						return new CssParser({
-							allowPseudoBlocks: false,
-							allowModeSwitch: false
+				for (const type of ["css", "css/global", "css/module"]) {
+					normalModuleFactory.hooks.createParser
+						.for(type)
+						.tap(plugin, parserOptions => {
+							validateParserOptions(parserOptions);
+
+							switch (type) {
+								case "css":
+									return new CssParser();
+								case "css/global":
+									return new CssParser({
+										allowPseudoBlocks: false,
+										allowModeSwitch: false
+									});
+								case "css/module":
+									return new CssParser({
+										defaultMode: "local"
+									});
+							}
+						});
+					normalModuleFactory.hooks.createGenerator
+						.for(type)
+						.tap(plugin, generatorOptions => {
+							validateGeneratorOptions(generatorOptions);
+							return this._exportsOnly
+								? new CssExportsGenerator()
+								: new CssGenerator();
 						});
-					});
-				normalModuleFactory.hooks.createParser
-					.for("css/module")
-					.tap(plugin, parserOptions => {
-						validateParserOptions(parserOptions);
-						return new CssParser({
-							defaultMode: "local"
+					normalModuleFactory.hooks.createModuleClass
+						.for(type)
+						.tap(plugin, (createData, resolveData) => {
+							if (resolveData.dependencies.length > 0) {
+								const dependency = resolveData.dependencies[0];
```

Comment:
Is there ever more than one dependency? (I don't know this logic very well but want to just cover the logic I'm reading here). How do we cover more than one dependency)? Should we add a check for ensuring just 1 if only 1 should be there?

Reply:
For CSS `@import` there is only one dependency (same logic is living in mini-css-extract-plugin a long time), maybe in future we will have more when `CSS` will support mutliple steelsheets in one CSS file, but it is only under dicussions in w3